### PR TITLE
Vault TLS auth

### LINKED ISSUE #1474: vault: enable TLS auth backend
This would make it way easier for machines in our cluster to interact with Vault, because they could authenticate with information they already have. Examples from the [backend's documentation](https://www.vaultproject.io/docs/auth/cert.html):

```bash
$ vault auth -method=cert -ca-cert=/etc/pki/CA/ca.cert -client-cert=/etc/pki/mantl/cert -client-key=/etc/pki/mantl/key
```

```bash
$ curl -X POST --cacert /etc/pki/CA/ca.cert --cert /etc/pki/mantl/cert --key /etc/pki/mantl/key 127.0.0.1:8200/v1/auth/cert/login
```

Work on this would obviously be impacted by/rely on #1463.

### PR DESCRIPTION
**Pull request by siddharthist**
- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes

fixes #1474 

since the build actually includes authenticating nodes with their certs (for sanity checking), this should be safe to merge as soon as the CI build is green.

### CODE REVIEW

**Comment by siddharthist** [comment]
builds failing at
```
TASK: [vault | set vault_token fact] *******************************************
fatal: [mantl-ci-931-1-control-02] => One or more undefined variables: 'vault_root_token' is undefined
fatal: [mantl-ci-931-1-control-03] => One or more undefined variables: 'vault_root_token' is undefined
fatal: [mantl-ci-931-1-control-01] => One or more undefined variables: 'vault_root_token' is undefined
```

File: roles/vault/tasks/main.yml [hunk 1]
```diff
@@ -45,3 +45,12 @@
     vault
 
 - include: bootstrap.yml
+
+- name: set vault_token fact
+  set_fact:
+    vault_token: "{{ vault_root_token }}"
+  when: >
+    "{{ vault_root_token is defined
+    and 'Vault is already initialized' not in vault_init_output.stdout }}"
```
Comment (ChrisAubuchon) [inline roles/vault/tasks/main.yml]: I think this needs to be:
```
and 'Vault is already initialized' in vault_init_output.stdout }}"
```
The `extract root token` task handles the `not in` case.

Comment (siddharthist) [inline roles/vault/tasks/main.yml]: oof, duh. Thanks!

**Comment by siddharthist** [comment]
one quota error, one calico error, one #1625 :disappointed: 

@nigerianlegend could you test this branch?

**Comment by ryane** [comment]
this is working for me:

```shell
vault auth -method=cert -ca-cert=/etc/pki/CA/ca.cert -client-cert=/etc/pki/mantl/cert -client-key=/etc/pki/mantl/key
Successfully authenticated! You are now logged in.
The token below is already saved in the session. You do not
need to "vault auth" again with the token.
token: <token>
token_duration: 2591999
token_policies: [default]
```

Question though: is it expected to be up to the user to write a policy and then update the cert to associate it with that policy? currently, it looks like the cert is only associated with the default policy which, afaict, doesn't have permissions to do anything in vault. seems a secure default but should we add something about that to the docs?

**Comment by ryane** [comment]
also, `vault_command_options` is out of sync in the readme for the role.

**Comment by siddharthist** [comment]
@ryane That's right. This is only a foundation upon which to build. Ultimately, the policy will be that nodes can renew their own TLS certificates by authenticating with the old ones.

**Comment by ryane** [comment]
lgtm

**Pull request merged**